### PR TITLE
upload/download file

### LINKED ISSUE #17: As a user, I should be able to upload a File to Hedera
- [x] InterfaceUploadFile
 - [x] Checkbox to "hash" file before uploading
 - [x] Success dialog should look like "CreateAccount" and show/require-to-copy the file ID

### PR DESCRIPTION
**Pull request by Sheng-Long**
Created upload component and view, blocked on SDK upload file support

Still need drag n drop support, sdk integration and hashing

closes #17 #18 #72

### CODE REVIEW

File: src/components/DownloadFile.vue [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+                fileLink.value.download =
+                    "MHW_File" + (type ? "." + type.ext : "");
+
+                context.root.$el.append(fileLink.value as Node);
+                fileLink.value.click();
+                context.root.$el.removeChild(
+                    fileLink.value as HTMLAnchorElement
+                );
+            } catch (error) {
+                console.log(error);
```
[the remark this answers is not in the document]
Comment (Sheng-Long) [reply src/components/DownloadFile.vue]: added alert which spits out the error

File: src/components/UploadFile.vue [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+                    FileCreateTransaction,
+                    Client,
+                    TransactionReceipt
+                } = await (import("@hashgraph/sdk") as Promise<
+                    typeof import("@hashgraph/sdk")
+                >);
+
+                const receipt = ref<TransactionReceipt.AsObject | null>(null);
+
+                receipt.value = await new FileCreateTransaction(
```
[the remark this answers is not in the document]
Comment (Sheng-Long) [reply src/components/UploadFile.vue]: don't think this is an issue anymore?

File: src/components/UploadFile.vue [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+                    .setExpirationTime(Date.now() + 7890000000)
+                    .addKey(store.state.wallet.session.privateKey.publicKey)
+                    .setTransactionFee(1e12)
+                    .build()
+                    .executeForReceipt();
+
+                if (receipt == null) {
+                    throw new Error("transaction returned is null");
+                }
+                console.log(typeof receipt.value.fileId.file);
```
[the remark this answers is not in the document]
Comment (Sheng-Long) [reply src/components/UploadFile.vue]: removed all console.logs in views and components relating to upload/download

File: src/components/UploadFile.vue [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+    width: 100%;
+}
+
+.upload-file {
+    align-items: center;
+    display: flex;
+    flex-direction: column;
+}
+
+.fileform {
```
Comment (QuestofIranon) [inline src/components/UploadFile.vue]: This selector also isn't used

File: src/components/UploadFile.vue [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+                    .setContents(state.fileUint8Array as Uint8Array)
+                    .setExpirationTime(Date.now() + 7890000000)
+                    .addKey(
+                        (await store.state.wallet.session.wallet.getPublicKey()) as import("@hashgraph/sdk").Ed25519PublicKey
+                    )
+                    .setTransactionFee(1e12)
+                    .build()
+                    .executeForReceipt();
+
+                if (receipt == null) {
```
Comment (QuestofIranon) [inline src/components/UploadFile.vue]: This if always resolves to false.

File: src/components/DownloadFile.vue [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+.download-container {
+    align-items: center;
+    border-radius: 5px;
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+}
+
+.fileform {
```
Comment (QuestofIranon) [inline src/components/DownloadFile.vue]: this selector isn't used

File: src/assets/file-document-box-multiple-outline.svg [hunk 7]
```diff
@@ -0,0 +1 @@
+<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd"><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" width="24" height="24" viewBox="0 0 24 24"><path d="M16,15H9V13H16V15M19,11H9V9H19V11M19,7H9V5H19V7M3,5V21H19V23H3A2,2 0 0,1 1,21V5H3M21,1A2,2 0 0,1 23,3V17C23,18.11 22.11,19 21,19H7A2,2 0 0,1 5,17V3C5,1.89 5.89,1 7,1H21M7,3V17H21V3H7Z" /></svg>
```
Comment (QuestofIranon) [inline src/assets/file-document-box-multiple-outline.svg]: Can you run a formatter on this? (I usually use auto format in Web Storm, not sure if VS Code has something)

File: src/assets/file-document-box-multiple.svg [hunk 8]
```diff
@@ -0,0 +1 @@
+<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd"><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" width="24" height="24" viewBox="0 0 24 24"><path d="M16,15H9V13H16M19,11H9V9H19M19,7H9V5H19M21,1H7C5.89,1 5,1.89 5,3V17C5,18.11 5.9,19 7,19H21C22.11,19 23,18.11 23,17V3C23,1.89 22.1,1 21,1M3,5V21H19V23H3A2,2 0 0,1 1,21V5H3Z" /></svg>
```
Comment (QuestofIranon) [inline src/assets/file-document-box-multiple.svg]: And on this?

File: src/components/DownloadFile.vue [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+
+        // eslint-disable-next-line unicorn/consistent-function-scoping
+        async function handleDownloadClick(): Promise<void> {
+            if (!store.state.wallet.session) {
+                throw new Error("session should not be null");
+            }
+            const client = store.state.wallet.session.client;
+            client.setMaxQueryPayment(1000000000000);
+            try {
+                // state.isBusy = true;
```
Comment (QuestofIranon) [inline src/components/DownloadFile.vue]: delete this comment

File: src/components/UploadFile.vue [hunk 10]
```diff
@@ -0,0 +1,286 @@
...
+//! end remove bit
+
+export default createComponent({
+    components: {
+        Button,
+        MaterialDesignIcon
+    },
+
+    setup(props, context) {
+        // const highlight = false;  //for dragndrop
```
Comment (QuestofIranon) [inline src/components/UploadFile.vue]: Delete this comment

File: src/components/UploadFile.vue [hunk 11]
```diff
@@ -0,0 +1,286 @@
...
+            }
+            if (hash.value && hash.value.checked) {
+                const digest = await crypto.subtle.digest(
+                    "SHA-384",
+                    state.fileUint8Array as Uint8Array
+                );
+                state.fileUint8Array = new Uint8Array(digest);
+            }
+
+            //TODO [2019-10-18]: Handle file splitting: if upload fails with some error which indirectly means: "file too big", chop file in half and try again with create for first half and fileAppendTransaction for second half - recursively do these halves until no error
```
Comment (QuestofIranon) [inline src/components/UploadFile.vue]: Can you make an issue for this if it's a future `todo`?

File: src/components/UploadFile.vue [hunk 12]
```diff
@@ -0,0 +1,286 @@
...
+                    });
+
+                    reader.readAsArrayBuffer(file);
+                }
+            );
+
+            target.value = ""; // change back to initial state to gaurantee that click fires next time
+            state.fileUint8Array = new Uint8Array(fileBuff);
+        }
+        //TODO [2019-10-18]: Should combine these functions
```
Comment (QuestofIranon) [inline src/components/UploadFile.vue]: This one also (make an issue)

File: src/assets/file-document-box-multiple-outline.svg (hunk 7 above)
Comment (Sheng-Long) [reply src/assets/file-document-box-multiple-outline.svg]: should we be?  A minified object like this will load slightly faster

Comment (Sheng-Long) [reply src/assets/file-document-box-multiple-outline.svg]: https://victorzhou.com/blog/minify-svgs/

File: src/components/UploadFile.vue (hunk 12 above)
Comment (Sheng-Long) [reply src/components/UploadFile.vue]: i just fixed this one

**Pull request merged**